### PR TITLE
Avoid snatch proper when codec is different from current proper

### PR DESCRIPTION
**Pull request by fernandog**
- [x] PR is based on the DEVELOP branch
- [x] Don't send big changes all at once. Split up big PRs into multiple smaller PRs that are easier to manage and review
- [x] Read [contribution guide](https://github.com/PyMedusa/SickRage/blob/master/.github/CONTRIBUTING.md)

### CODE REVIEW

**Comment by fernandog** [comment]
video codec is working:
FINDPROPERS :: [b8b258a] Parsed Fear.the.Walking.Dead.S02E11.PROPER.HDTV.x264-KILLERS[rartv] into title: Fear the Walking Dead, season: 2, episode: [11], other: Proper, proper_count: 1, proper_tag: PROPER, format: HDTV, video_codec: h264, release_group: KILLERS, type: episode, absolute_episode: [], quality: SDTV

**Comment by fernandog** [comment]
need to get release_name from history
only status `downloaded` has `release_name`

**Comment by medariox** [comment]
What if the user clears cache?
What is the problem if only downloaded has release_name, isn't that all we need?

**Comment by fernandog** [comment]
@medariox done, user can clear history and check will continue to work

File: sickbeard/properFinder.py [hunk 1]
```diff
@@ -237,24 +243,37 @@ def _get_proper_results(self):  # pylint: disable=too-many-locals, too-many-bran
...
+                current_codec = NameParser()._parse_string(release_name).video_codec
+                # Ignore proper if codec differs from downloaded release codec
+                if all([current_codec, parse_result.video_codec, parse_result.video_codec != current_codec]):
+                    logger.log('Ignoring proper because codec is different: {name}'.format(name=best_result.name))
+                    self.processed_propers.append(cur_proper.name)
+                    continue
+            else:
+                # We shouldn't get here unless some DB error
+                logger.log("Something is wrong! Status is 'Downloaded' and we don't have a release name. "
+                           "Couldn't compare codec", logger.ERROR)
```
Comment (medariox) [inline sickbeard/properFinder.py]: Did you check your DB? I have a few of these in my DB... don't think it's safe to make this ERROR.

Comment (medariox) [inline sickbeard/properFinder.py]: To replicate it: snatch an episode, before the download finishes, set it to downloaded. We should probably check if there is a file there for that episode, before changing it to download. Anyway, right now we allow this behavior.

**Comment by medariox** [comment]
Approved

**Comment by fernandog** [comment]
Fixes https://github.com/pymedusa/SickRage/issues/532 by not snatching new releases until episode get status `downloaded`

**Pull request merged**